src/rpc/net.cpp

```diff
@@ -544,16 +544,20 @@ UniValue setban(const JSONRPCRequest& request)
             absolute = true;
 
         if (isSubnet) {
-            g_connman->Ban(subNet, BanReasonManuallyAdded, banTime, absolute);
-            g_connman->DisconnectNode(subNet);
+            g_banman->Ban(subNet, BanReasonManuallyAdded, banTime, absolute);
+            if (g_connman) {
+                g_connman->DisconnectNode(subNet);
+            }
         } else {
-            g_connman->Ban(netAddr, BanReasonManuallyAdded, banTime, absolute);
-            g_connman->DisconnectNode(netAddr);
+            g_banman->Ban(netAddr, BanReasonManuallyAdded, banTime, absolute);
+            if (g_connman) {
```

Comment:
Commit [banman: create and split out banman].

Perhaps pull the common `DisconnectNode` code out of this `if/else` statement?

Reply:
One is a subnet, the other is an address. CSubNet is a subclass of CNetAddr, so we could potentially store a generic one, but imo duplicating the disconnect is more straightforward.